clio-client/src/main/scala/org/broadinstitute/clio/client/commands/CommandDispatch.scala

```diff
@@ -3,31 +3,30 @@ package org.broadinstitute.clio.client.commands
 import akka.http.scaladsl.model.HttpResponse
 import com.typesafe.scalalogging.LazyLogging
 import org.broadinstitute.clio.client.ClioClientConfig
-import org.broadinstitute.clio.client.commands.Commands.{
-  AddWgsUbam,
-  QueryWgsUbam
-}
-import org.broadinstitute.clio.client.parser.BaseArgs
 import org.broadinstitute.clio.client.webclient.ClioWebClient
 
 import scala.concurrent.{Await, ExecutionContext, Future}
 
 object CommandDispatch extends LazyLogging {
 
-  def execute(command: CommandType, webClient: ClioWebClient, config: BaseArgs)(
-    implicit ec: ExecutionContext
-  ): Future[HttpResponse] = {
+  def execute(
+    command: CommandType,
+    webClient: ClioWebClient,
+    bearerToken: String
+  )(implicit ec: ExecutionContext): Future[HttpResponse] = {
     command match {
-      case AddWgsUbam   => AddWgsUbamCommand.execute(webClient, config)
-      case QueryWgsUbam => QueryWgsUbamCommand.execute(webClient, config)
+      case commandType: AddWgsUbam =>
+        new AddWgsUbamExecutor(commandType)
+          .execute(webClient, bearerToken)
+      case commandType: QueryWgsUbam =>
```

Comment:
Is a way to make `CommandType` return the correct constructor to call, to avoid using `match` here? Or maybe we aren't trying to avoid using match in our scala code?

Reply:
It was suggested that all execute code is handled by the dispatcher since CommandType is a sealed trait and we didn't want the commands themselves to get too big.

Reply:
OK, I remember from the scala class that using `match` for `trait`s was the scala way.